axelor-account/src/main/java/com/axelor/apps/account/service/analytic/AnalyticMoveLineServiceImpl.java

```diff
@@ -229,83 +228,40 @@ public AnalyticMoveLine computeAnalyticMoveLine(
           .divide(new BigDecimal(100), RETURN_SCALE, RoundingMode.HALF_UP);
     }
     return BigDecimal.ZERO;
   }
 
   @Override
-  public BigDecimal getAnalyticAmountFromParent(Context parent, AnalyticMoveLine analyticMoveLine) {
-    if (MoveLine.class.equals(parent.getContextClass())) {
-      MoveLine line = parent.asType(MoveLine.class);
-      if (analyticMoveLine != null && line != null) {
-        return getAnalyticAmount(line, analyticMoveLine);
-      }
-    } else if (InvoiceLine.class.equals(parent.getContextClass())) {
-      InvoiceLine line = parent.asType(InvoiceLine.class);
-      if (analyticMoveLine != null && line != null) {
-        return getAnalyticAmount(line, analyticMoveLine);
-      }
-    }
-    return BigDecimal.ZERO;
-  }
-
-  @Override
-  public AnalyticJournal getAnalyticJournalFromParent(Context parent) throws AxelorException {
-    if (MoveLine.class.equals(parent.getContextClass())) {
-      MoveLine line = parent.asType(MoveLine.class);
-      if (line.getAccount() != null && line.getAccount().getCompany() != null) {
-        return accountConfigService
-            .getAccountConfig(line.getAccount().getCompany())
-            .getAnalyticJournal();
-      }
-    } else if (InvoiceLine.class.equals(parent.getContextClass())) {
-      InvoiceLine line = parent.asType(InvoiceLine.class);
-      if (line.getAccount() != null && line.getAccount().getCompany() != null) {
-        return accountConfigService
-            .getAccountConfig(line.getAccount().getCompany())
-            .getAnalyticJournal();
-      }
+  public AnalyticJournal getAnalyticJournalFromParent(AnalyticLine line) throws AxelorException {
+    if (line.getAccount() != null && line.getAccount().getCompany() != null) {
+      return accountConfigService
+          .getAccountConfig(line.getAccount().getCompany())
+          .getAnalyticJournal();
     }
     return null;
   }
 
   @Override
-  public LocalDate getDateFromParent(Context parent) {
-    if (MoveLine.class.equals(parent.getContextClass())) {
-      MoveLine line = parent.asType(MoveLine.class);
+  public LocalDate getDateFromParent(AnalyticLine parent) {
+    if (parent instanceof MoveLine) {
+      MoveLine line = (MoveLine) parent;
       if (line.getDate() != null) {
         return line.getDate();
-      } else if (line.getAccount() != null && line.getAccount().getCompany() != null) {
-        return appBaseService.getTodayDate(line.getAccount().getCompany());
-      }
-    } else if (InvoiceLine.class.equals(parent.getContextClass())) {
-      InvoiceLine line = parent.asType(InvoiceLine.class);
-      if (line.getAccount() != null && line.getAccount().getCompany() != null) {
-        return appBaseService.getTodayDate(line.getAccount().getCompany());
       }
     }
+    if (parent.getAccount() != null && parent.getAccount().getCompany() != null) {
+      return appBaseService.getTodayDate(parent.getAccount().getCompany());
+    }
     return LocalDate.now();
```

Comment:
appBaseService.getTodayDate(Company company) is a method that can be called if company is null.
So LocalDate.now() should not be called but appBaseService should also be used.